rasa/cli/x.py

```diff
@@ -291,16 +304,76 @@ def rasa_x(args: argparse.Namespace):
         run_locally(args)
 
 
-def run_in_production(args: argparse.Namespace):
-    from rasa.cli.utils import print_success
-    from rasa.core.utils import AvailableEndpoints
+def _pull_runtime_config_from_server(
+    config_endpoint: Optional[Text],
+    attempts: int = 30,
+    wait_time_between_pulls: Union[int, float] = 0.5,
+    keys: Tuple[Text, ...] = ("endpoints", "credentials"),
+) -> Tuple[Text, ...]:
+    """Pull runtime config from `config_endpoint`.
 
-    print_success("Starting Rasa X in production mode... 🚀")
+    Returns a path to yaml dumps, each containing the contents of one of `keys`.
+    """
+
+    while attempts > 0:
+        try:
+            response = requests.get(config_endpoint)
+            if response.status_code == 200:
+                rjs = response.json()
+                return tuple(_dump_dict_to_yaml(rjs, k) for k in keys)
```

Comment:
Not sure whether a list isn't fine instead of a tuple 🤔

Reply:
we can definitely use a list in the return